zkevm-circuits/src/evm_circuit/execution/logs.rs

```diff
@@ -205,28 +203,27 @@ impl<F: Field> ExecutionGadget<F> for LogGadget<F> {
         self.memory_expansion
             .assign(region, offset, step.memory_word_size(), [memory_address])?;
 
-        let opcode = step.opcode.unwrap();
+        let opcode = step.opcode().unwrap();
         let topic_count = opcode.postfix().expect("opcode with postfix") as usize;
         assert!(topic_count <= 4);
 
         let is_persistent = call.is_persistent as u64;
-        let mut topic_stack_entry = if topic_count > 0 {
-            step.rw_indices[6 + call.is_persistent as usize]
-        } else {
-            // if topic_count == 0, this value will be no used anymore
-            (RwTableTag::Stack, 0usize)
-        };
+        let topic_stack_entry =
+            (topic_count > 0).then(|| step.rw_index(6 + call.is_persistent as usize));
```

Comment:
I think this can be explained by looking at the rw operations that the log does. See https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/14bd7c9ca95a5b0ed50d78750414e8c11f799e8e/bus-mapping/src/evm/opcodes/logs.rs#L53

- 0-1. Stack read
- 2-5. call context read
- 6? tx log write
- 6 / 7. Stack read